kivy/input/providers/mouse.py

```diff
@@ -293,6 +304,12 @@ def on_mouse_release(self, win, x, y, button, modifiers):
             self.alt_touch = None
         return True
 
+    def on_mouse_pos(self, win, mouse_pos):
+        width, height = EventLoop.window.system_size
+        args = (mouse_pos[0] / width, mouse_pos[1] / height)
+        event = MouseHoverEvent(self.device, None, args)
```

Comment:
Would if be possible to have only one event at all for the mouse ? If you look at MouseMotionEvent, it is created everytime the user touch down, and deleted when the user touch up. In your case, we could have a single hover event for the whole time of the app (for the mouse event provider).

Reply:
We will need at least two events so that we can send 'end' argument to handle two widgets overlapping case. I think we have cases where touch event is kept in widgets instance like a last touch, so maybe there is a case where hover event is also needed to be kept. And we still need to handle multi-hover which also needs to work correctly for overlapping case.

Reply:
I would have to think more about single event per provider or cache of events per provider.

Reply:
Yeah, I kinda agree that reusing a hover event, while maybe performant would lead to usability issues because people will expect they can save the event, like last touch event and maybe manipulate it after the event is dispatched.